esp-hal/src/soc/esp32/gpio.rs

```diff
@@ -61,11 +61,16 @@ pub const NUM_PINS: usize = 40;
 
 pub(crate) const FUNC_IN_SEL_OFFSET: usize = 0;
 
+#[doc(hidden)]
```

Comment:
This applies to all changes in the `soc/$CHIP/gpio.rs` file: do these types actually need to be public? Can they be `pub(crate)` instead? I would really prefer not to litter too many `#[doc(hidden)]` attributes throughout the code base, it usually implies something is not quite correct.

Reply:
Updated!